src/components/BalanceChart/BalanceChart.tsx

```diff
@@ -28,13 +31,24 @@ export const BalanceChart: React.FC<BalanceChartArgs> = ({
     timeframe,
   })
 
+  const priceHistoryDataEmpty = useAppSelector(state =>
+    selectPriceHistoriesEmptyByAssetTimeframe(state, assetIds, timeframe),
+  )
+
   useEffect(
     () => setPercentChange(calculatePercentChange(balanceChartData)),
     [balanceChartData, setPercentChange],
   )
 
   const color = percentChange > 0 ? 'green.500' : 'red.500'
 
+  if (!balanceChartDataLoading && priceHistoryDataEmpty.length === assetIds.length)
```

Comment:
```suggestion
  if (!balanceChartDataLoading && isPriceHistoryDataEmpty)
```

there's no need for this logic in the view layer if the selector can do the logic. views are for displaying things.